Content.Server/Explosion/TriggerSystem.cs

```diff
@@ -116,5 +143,23 @@ public void HandleTimerTrigger(TimeSpan delay, IEntity triggered, IEntity? user
                 Trigger(triggered, user);
             });
         }
+        public void SetProximityFixture(EntityUid uid, TriggerOnProximityComponent component, bool remove)
+        {
+            var entity = EntityManager.GetEntity(uid);
+            var broadphase = Get<SharedBroadphaseSystem>();
+
+            if (entity.TryGetComponent(out PhysicsComponent? physics))
+            {
+                var fixture = physics.GetFixture(component.ProximityFixture);
+                if (!remove && fixture != null)
+                {
+                    broadphase.DestroyFixture(physics, fixture);
+                }
+                else
+                {
+                    broadphase.CreateFixture(physics, new Fixture(physics, component.Shape) { CollisionLayer = (int) (CollisionGroup.MobImpassable | CollisionGroup.SmallImpassable | CollisionGroup.SmallImpassable), Hard = false, ID = component.ProximityFixture });
+                }
```

Comment:
this will remove a fixture and not create a new one if one already exists and this is called with remove set to false. i think you want something like this
```suggestion
                if (remove ^ fixture != null) return;
                if (remove)
                {
                    broadphase.DestroyFixture(physics, fixture);
                }
                else
                {
                    broadphase.CreateFixture(physics, new Fixture(physics, component.Shape) { CollisionLayer = (int) (CollisionGroup.MobImpassable | CollisionGroup.SmallImpassable | CollisionGroup.SmallImpassable), Hard = false, ID = component.ProximityFixture });
                }
```